### PR TITLE
grpc, pythonPackges.grpcio, pythonPackages.grpcio-tools: update to 1.16.1

### PR DESCRIPTION
**Pull request by lopsided98**
###### Motivation for this change

Updates grpc and related packages to 1.16.1:
* grpc: 1.15.0 -> 1.16.1
* pythonPackages.grpcio: 1.15.0 ->1.16.1
* pythonPackages.grpcio-tools: 1.14.2 -> 1.16.1

I removed the grpc dependency from grpcio-tools, because it is not used.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build grpc pythonPackages.grpcio python3Packages.grpcio pythonPackages.grpcio-tools python3Packages.grpcio-tools python3Packages.tensorflow pythonPackages.tensorflow

**Comment by lopsided98** [comment]
I added a commit which applies protocolbuffers/protobuf#4862 to make protobuf compatible with Python 3.7.

File: pkgs/development/python-modules/protobuf/default.nix [hunk 1]
```diff
@@ -16,6 +16,15 @@ buildPythonPackage rec {
   propagatedBuildInputs = [ protobuf google_apputils ];
   buildInputs = [ google_apputils pyext ];
 
+  patches = [
+    # Python 3.7 compatibility (remove when protobuf 3.7 is released)
+    (fetchpatch {
+      url = "https://github.com/protocolbuffers/protobuf/pull/4862.patch";
```
[the remark this answers is not in the document]
Comment (lopsided98) [reply pkgs/development/python-modules/protobuf/default.nix]: Fixed

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build grpc pythonPackages.grpcio python3Packages.grpcio pythonPackages.grpcio-tools python3Packages.grpcio-tools pythonPackages.protobuf python3Packages.protobuf

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build pythonPackages.tensorflow

**Comment by worldofpeace** [comment]
Seems `pythonPackages.tensorflow` has been broken on aarch64-linux for a while now https://hydra.nixos.org/job/nixpkgs/trunk/python27Packages.tensorflow.aarch64-linux/all

**Review by worldofpeace — APPROVED** [review]
All usual checks pass for me. Thanks for picking this up :+1:

**Comment by samueldr** [comment]
The protobuf patch introduced here does not apply when using another (older) protobuf (e.g. protobuf_3_1).

```
$ nix-build -E 'with import ./. { }; python3Packages.protobuf.override { protobuf = protobuf3_1; }'
unpacking sources
unpacking source archive /nix/store/1zdyl0cxaa8ha2v1zp75zzdjd6j99d0m-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/util/python/BUILD
patching sources
applying patch /nix/store/yagx7hvylnnjq7lxbcia0y5lq1r736w3-0a59054c30e4f0ba10f10acfc1d7f3814c63e1a7.patch
patching file google/protobuf/pyext/descriptor.cc
Hunk #1 succeeded at 55 (offset -1 lines).
patching file google/protobuf/pyext/descriptor_containers.cc
patching file google/protobuf/pyext/descriptor_pool.cc
Hunk #1 succeeded at 47 (offset -1 lines).
patching file google/protobuf/pyext/extension_dict.cc
Hunk #1 FAILED at 53.
1 out of 1 hunk FAILED -- saving rejects to file google/protobuf/pyext/extension_dict.cc.rej
patching file google/protobuf/pyext/message.cc
Hunk #1 succeeded at 82 (offset 3 lines).
Hunk #2 succeeded at 1425 (offset -104 lines).
```

(PR incoming)

**Pull request merged**